src/transformStateWithClones.js

```diff
@@ -6,8 +6,55 @@
  *
  * @return {Object[]}
  */
+/* eslint no-console: ["error", { allow: ["warn", "error", "log"] }] */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  for (const item of actions) {
+    if (item.type === 'addProperties') {
+      // if our 'result' arr has no therefore we use state obj
+      if (result.length === 0) {
+        const objWithAddedProps = Object.assign({}, state, item.extraData);
+
+        result.push(objWithAddedProps);
+      } else {
+        // if our 'result' arr has last item then we use last item
+        const objWithAddedProps = Object.assign(
+          {},
+          result[result.length - 1],
+          item.extraData,
+        );
```

Comment:
[DONT REPEAT YOURSELF] - There's repeated logic for creating a new object with added properties. This can be simplified by performing the action once after the switch statement.